client/main/dataset.js

```diff
@@ -810,30 +810,32 @@ const DataSetModel = Backbone.Model.extend({
                 }
                 else if (change.hiddenChanged || change.created) {
                     let column = this.getColumnById(change.id);
-                    if (column.hidden) {
-                        if (change.dIndex > old.right) {  // to the right of the view
-                            // do nothing
-                        }
-                        else if (change.dIndex < old.left) {  // to the left of the view
-                            viewport.left  -= 1;
-                            viewport.right -= 1;
-                        }
-                        else {
-                            viewport.right -= 1;
-                        }
-                    }
-                    else {
-                        if (column.dIndex > viewport.right) {  // to the right of the view
-                            // do nothing
-                        }
-                        else if (column.dIndex < viewport.left) {
-                            viewport.left  += 1;
-                            viewport.right += 1;
+                    if ( ! (change.created && column.hidden)) {
```

Comment:
this if-statement is doing my head in ... i suppose it's equivalent to `if ( ! change.created || ! column.hidden) {`

so we've got this?:

```
if  ( ! change.created || ! column.hidden) {
    if (column.hidden) {
        // doA();
    }
    else {
        // doB();
    }
}
```

can we rewrite this more clearly as?:

```
if ( ! change.created) {
    // doA();
}
else if ( ! column.hidden) {
    // doB();
}
```

Reply:
No that's not quiet right. This is one reason why i left it the way i did because it was the easiest without changing to much. 

`if ( ! change.created && column.hidden) {
    // doA();
}
else if ( ! column.hidden) {
    // doB();
}`

I think this works

Reply:
oh yup. i think that's pretty clear.

Reply:
wait, no. i think this *is* correct:

```
if ( ! change.created && column.hidden) {
```

but it *isn't* equivalent to the original:

```
if ( ! (change.created && column.hidden)) {
```

so i think the original isn't correct ... ?

Reply:
oh wait, no, it probably is

Reply:
hehehe

This is why i left it as the over arching gate-keeper... :) confidence

Reply:
but i'm pretty sure that 

```
if ( ! change.created && column.hidden) {
    // doA();
}
else if ( ! column.hidden) {
    // doB();
} 
```

is correct